migrations/cavil.sql

```diff
@@ -215,3 +215,31 @@ create index on snippets(approved);
 
 -- 6 down
 drop index snippets_approved_idx if exists;
+
+-- 7 up
+alter table license_patterns add risk int not null default 5;
+alter table license_patterns add eula boolean not null default false;
+alter table license_patterns add nonfree boolean not null default false;
+
+-- 7 down
+alter table license_patterns drop column risk, drop column eula, drop column nonfree;
+
+-- 8 up
+alter table license_patterns drop column license;
+
+-- 8 down
+alter table license_patterns add column license int not null references licenses (id);
+
+-- 9 up
+alter table license_patterns alter column license_string set default '';
+alter table license_patterns alter column license_string set not null;
+
+-- 9 down
+alter table license_patterns alter column license_string drop default;
+alter table license_patterns alter column license_string set null;
+
+-- 10 up
+alter table license_patterns rename column license_string to license;
+
+-- 10 down
+alter table license_patterns rename column license to license_string;
```

Comment:
You probably should make those one migration before merging. 😉 It doesn't matter if 7, 8 and 9 are not present as long as the latest number is still there, so those could just be merged into 6 for example.

Reply:
I need to migrate the data before dropping the license key. So I can't have them in one migration step.